### PR TITLE
Update seen async errors

### PR DESCRIPTION
**Pull request by kk1050**
# ARTIQ Pull Request

## Description of Changes
Update the SEEN_ASYNC_ERRORS and add unsafe to it 
### Related Issue
 destination_survey should update SEEN_ASYNC_ERRORS #1895 

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

### All Pull Requests

- [x] Use correct spelling and grammar.
- [x] Close/update issues.
- [x] Check the copyright situation of your changes and sign off your patches (`git commit --signoff`, see [copyright](../CONTRIBUTING.rst#copyright-and-sign-off)).

### Code Changes

- [x] Test your changes or have someone test them. Mention what was tested and how.

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

**Comment by sbourdeauducq** [comment]
What did you test?

**Comment by kk1050** [comment]
> What did you test?

I build it and connect the satellite to the master but i can't test the errors.

**Comment by sbourdeauducq** [comment]
You can test the errors, it's pretty straightforward to generate all types by running specifically crafted kernels.

File: artiq/firmware/runtime/rtio_mgt.rs [hunk 1]
```diff
@@ -212,11 +212,11 @@ pub mod drtio {
                                 destination_set_up(routing_table, up_destinations, destination, false),
                             Ok(drtioaux::Packet::DestinationOkReply) => (),
                             Ok(drtioaux::Packet::DestinationSequenceErrorReply { channel }) =>
-                                error!("[DEST#{}] RTIO sequence error involving channel 0x{:04x}", destination, channel),
+                                unsafe{SEEN_ASYNC_ERRORS |= 4},
```
Comment (sbourdeauducq) [inline artiq/firmware/runtime/rtio_mgt.rs:215]: Those numbers are also used here:
https://github.com/m-labs/artiq/blob/9b0bdd1ce32666ff928b2a52953298409118b613/artiq/firmware/runtime/rtio_mgt.rs#L160-L168
Define some ``const``

Comment (sbourdeauducq) [reply artiq/firmware/runtime/rtio_mgt.rs:215]: And we still want the ``error!`` messages, the "seen errors" at the end of kernel execution is just an additional (and imprecise) indicator as discussed in https://github.com/m-labs/artiq/pull/1889.

**Pull request merged**